src/Razor/src/Microsoft.VisualStudio.Editor.Razor/Telemetry/TelemetryReporter.cs

```diff
@@ -85,20 +85,17 @@ public void ReportEvent(string name, Severity severity, params Property[] proper
 
     private static void AddToProperties(IDictionary<string, object?> properties, Property property)
     {
-        if (IsNumericValue(property.Value))
+        if (IsComplexValue(property.Value))
         {
-            properties.Add(GetPropertyName(property.Name), property.Value);
+            properties.Add(GetPropertyName(property.Name), new TelemetryComplexProperty(property.Value));
         }
         else
         {
-            properties.Add(GetPropertyName(property.Name), new TelemetryComplexProperty(property.Value));
+            properties.Add(GetPropertyName(property.Name), property.Value);
         }
 
-        static bool IsNumericValue(object? o)
-        {
-            return o is Type { IsEnum: false } type &&
-                   Type.GetTypeCode(type) is >= TypeCode.Char and <= TypeCode.Double;
-        }
+        static bool IsComplexValue(object? o)
+            => o is Type type && Type.GetTypeCode(type) == TypeCode.Object;
```

Comment:
It looks like this has a serious bug that will break telemetry. It should be `o?.GetType() is Type type`.

Reply:
I'm not sure how it's different than before that was changed in https://github.com/dotnet/razor/pull/9465

It looks like that was when the GetType call was removed

Reply:
Yup! It was my fault. Already submitted #9507 to fix my blunder.